metrics/test_driver/flutter_web_driver.dart

```diff
@@ -1,56 +1,142 @@
-  await FileUtils.downloadChromeDriver(workingDirPath);
-  await FileUtils.downloadFirefoxDriver(workingDirPath);
-
-  print('Running $seleniumFileName...');
-
-  await processManager.startSelenium(seleniumFileName, workingDirPath);
-
-  print("Selenium server is up, running flutter application...");
-
-  await processManager.startFlutterApp().catchError((error) {
-    exitCode = 1;
-    processManager.exitApp();
-  });
-
-  stdout.done.asStream().listen((_) => processManager.exitApp());
-
-  print("Application is up, running tests...");
-
-  await processManager
-      .startDriverTests(browserName: args.browserName)
-      .catchError((_) {
-    processManager.exitApp();
-  });
-
-  print(
-      "Flutter logs are stored in ${logsDir.absolute.uri}${ProcessManager.flutterLogsFileName}.log file");
-  print(
-      "Driver logs are stored in ${logsDir.absolute.uri}${ProcessManager.flutterLogsFileName}.log file");
-
-  processManager.exitApp();
+import 'arguments/model/driver_test_arguments.dart';
+import 'arguments/parser/driver_test_arguments_parser.dart';
+import 'cli/flutter/runner/flutter_drive_process_runner.dart';
+import 'cli/flutter/runner/flutter_run_process_runner.dart';
+import 'cli/selenium/runner/selenium_process_runner.dart';
+import 'cli/selenium/selenium.dart';
+import 'common/config/logs_file_config.dart';
+import 'common/logger/logger.dart';
+import 'process_manager/process_manager.dart';
+
+/// Runs the application and driver tests for this application.
+class FlutterWebDriver {
+  final DriverTestArguments _args;
+  final ProcessManager _processManager = ProcessManager();
+
+  /// Creates the [FlutterWebDriver].
+  ///
+  /// [args] is the application arguments to run with.
+  /// See [DriverTestArgumentsParser] to get all available arguments.
+  FlutterWebDriver(List<String> args)
+      : _args = DriverTestArgumentsParser.parseArguments(args);
+
+  /// Configures environment and starts driver tests.
+  Future<void> startDriverTests() async {
+    final bool verbose = _args.verbose;
+    final int port = _args.port;
+
+    await _prepareSelenium();
+
+    _setupLogger(_args.quiet);
+
+    _setupDispose();
+
+    Logger.log('Running selenium...');
+    await _processManager.run(
+      SeleniumProcessRunner(),
+      workingDir: _args.workingDir,
+    );
+
+    Logger.log("Selenium server is up, running flutter application...");
+    final flutterAppProcess = await _runFlutterApp(port, verbose);
+
+    Logger.log("Application is up, running tests...");
+    await _runDriverTests(port, verbose);
+    _processManager.kill(flutterAppProcess);
+
+    Logger.log('Running application using SKIA...');
+    await _runFlutterApp(port, verbose, useSkia: true);
+
+    Logger.log("SKIA Application is up, running tests...");
+    await _runDriverTests(port, verbose);
+
+    _tearDown();
+  }
+
+  /// Handles application termination and disposes the process manager.
```

Comment:
Is this what called on any error? Why don't we add a bit more information on why we need this and not what it does?